File: dp-core/src/main/java/com/dataplatform/org/entity/Department.java
package com.dataplatform.org.entity;

import lombok.Data;
import java.util.Date;
import java.util.List;

/**
 * 部门实体类
 * 
 * @author dataplatform
 */
@Data
public class Department {
    /** 部门ID */
    private Long id;
    
    /** 部门名称 */
    private String deptName;
    
    /** 部门编码 */
    private String deptCode;
    
    /** 父部门ID，0为顶级部门 */
    private Long parentId;
    
    /** 祖级列表（如 0,1,2） */
    private String ancestors;
    
    /** 负责人 */
    private String leader;
    
    /** 联系电话 */
    private String phone;
    
    /** 邮箱 */
    private String email;
    
    /** 排序 */
    private Integer sortOrder;
    
    /** 状态: 0-禁用 1-启用 */
    private Integer status;
    
    /** 删除标志: 0-正常 1-删除 */
    private Integer delFlag;
    
    /** 创建时间 */
    private Date createTime;
    
    /** 更新时间 */
    private Date updateTime;
    
    /** 子部门列表（非数据库字段） */
    private transient List<Department> children;
    
    /** 父部门名称（非数据库字段） */
    private transient String parentName;
}

File: dp-ui/src/api/org/department.ts
import request from '@/api/request'

export interface Department {
  id?: number
  deptName: string
  deptCode?: string
  parentId: number
  ancestors?: string
  leader?: string
  phone?: string
  email?: string
  sortOrder?: number
  status?: number
  createTime?: string
  updateTime?: string
  children?: Department[]
  parentName?: string
}

/** 获取部门列表（平铺） */
export const getDepartmentList = (params?: { keyword?: string; status?: number }) => {
  return request.get('/department/list', { params })
}

/** 获取部门树 */
export const getDepartmentTree = (params?: { keyword?: string; status?: number }) => {
  return request.get('/department/tree', { params })
}

/** 获取部门详情 */
export const getDepartmentById = (id: number) => {
  return request.get(`/department/${id}`)
}

/** 创建部门 */
export const createDepartment = (data: Department) => {
  return request.post('/department', data)
}

/** 更新部门 */
export const updateDepartment = (id: number, data: Department) => {
  return request.put(`/department/${id}`, data)
}

/** 删除部门 */
export const deleteDepartment = (id: number) => {
  return request.delete(`/department/${id}`)
}

/** 更新部门状态 */
export const updateDepartmentStatus = (id: number, status: number) => {
  return request.put(`/department/${id}/status`, { status })
}

/** 批量更新部门排序 */
export const batchUpdateDepartmentSort = (items: Array<{ id: number; sortOrder: number }>) => {
  return request.post('/department/batch-sort', items)
}
